File: terre09.rb
# Racine carrée d'un nombre

argument = ARGV

nombre = ARGV[0].to_i                                                   # variable nombre contenant l'argument, converti en integer
racinecarree = 1                                                

intervalle_resultat = nombre - 0.001                                    # je fixe l'intervalle de résultat approximatif du nombre donné

if argument.length != 1 || ARGV[0] =~ /\D/                              # si l'utilisateur n'entre rien ou + d'un argument ou autre chose que des nombres
    puts "erreur : insérer seulement un nombre"                         # affiche ce message d'erreur
else

    until (intervalle_resultat..nombre).include?(racinecarree ** 2)     # jusqu'à ce qu'un nombre au carré donne un résultat qui se situe dans l'intervalle 
        racinecarree += 0.0001                                          # ajoute 0.0001 à ce nombre
        resultat_approximatif = format("%.3f", racinecarree)            # affiche le nombre contenu dans racinecarree quand son resultat se situe dans l'intervalle, arrondi au millième 
    end

    if resultat_approximatif.to_f % 1 == 0                                              # si ce résultat approximatif est un entier
        puts "#{resultat_approximatif.sub!(/\.0+\z/, "")} : c'est un carré parfait !"   # affiche le résultat en enlevant les 0 après la virgule
    else
        puts "#{resultat_approximatif} : ceci est un résultat approximatif"             # sinon, affiche le résultat approximatif de la racine carrée  
    end

end


# j'attribue 1 à la variable racinecarree, afin de commencer à chercher la racine carrée dès le chiffre 1

File: terre09.js
// Racine carrée d'un nombre

let arguments = process.argv;

let number = process.argv[2];

if (!number) {
  console.log("erreur : insérez un nombre");
} else if (arguments.length > 3) {
  console.log("erreur : n'insérez qu'un nombre");
} else if (!/^\d+$/.test(number)) {
  console.log("erreur : n'insérez rien d'autre qu'un nombre");
} 

else {
  let perfectSquare = false;
  let i = 1;
  for (; i ** 2 <= number; i++) {
    if (i ** 2 == number) {
      console.log("La racine carrée de " + number + " est " + i + " c'est un carré parfait !");
      perfectSquare = true;
      break;
    }
  }
  
  if (!perfectSquare) {
    let j = i - 1;
    for (; j ** 2 <= number; j += 0.001) {
      if (j ** 2 > number - 0.001 && j ** 2 <= number) {
        jRound = j.toFixed(3);
        console.log("La racine carrée de " + number + " est approximativement " + jRound);
        break;
      }
    }
  }
}

